### PR TITLE
feat(OnyxIconButton): Implement OnyxIconButton

### PR DESCRIPTION
**Pull request by JoCa96**
Relates to #386 

Implemented OnyxIconButton

## Checklist

- [x] The added / edited code has been documented with [JSDoc](https://jsdoc.app/about-getting-started)
- [x] All changes are documented in the documentation app (folder `apps/docs`)
- [x] If a new component is added, at least one [Playwright screenshot test](https://github.com/SchwarzIT/onyx/actions/workflows/playwright-screenshots.yml) is added
- [x] A changeset is added with `npx changeset add` if your changes should be released as npm package (because they affect the library usage)

### CODE REVIEW

File: packages/sit-onyx/src/components/OnyxIconButton/OnyxIconButton.vue [hunk 1]
```diff
@@ -0,0 +1,112 @@
...
+    :aria-label="props.label"
+    :class="{
+      [`onyx-icon-button--${props.variation}`]: true,
+      'onyx-icon-button--loading': props.loading,
+    }"
+    :disabled="props.disabled || props.loading"
+    @click="emit('click')"
+  >
+    <OnyxLoadingIndicator v-if="props.loading" type="circle" />
+    <OnyxIcon v-else-if="props.icon" :icon="props.icon" size="24px" />
```
Comment (larsrickert) [inline packages/sit-onyx/src/components/OnyxIconButton/OnyxIconButton.vue:35]:
```suggestion
    <OnyxIcon v-else-if="props.icon" :icon="props.icon" />
```

24px is the default size. Should we add a property to set it? We could allow the same icon sizes as the regular OynxIcon

File: packages/sit-onyx/src/components/OnyxIconButton/OnyxIconButton.vue [hunk 2]
```diff
@@ -0,0 +1,112 @@
...
+import type { OnyxIconButtonProps } from "./types";
+
+const props = withDefaults(defineProps<OnyxIconButtonProps>(), {
+  disabled: false,
+  type: "button",
+  variation: "primary",
+});
+
+defineSlots<{
+  /** Slot for an custom icon */
```
Comment (larsrickert) [inline packages/sit-onyx/src/components/OnyxIconButton/OnyxIconButton.vue:13]:
```suggestion
  /** Slot for an custom icon. Property `icon` must be omitted if custom content is passed. */
```

**Review by larsrickert — CHANGES_REQUESTED** [review]

File: packages/sit-onyx/playwright/snapshots/components/OnyxIconButton/button-state--disabled-variation--secondary-focusState--active-chromium-linux.png
Comment (larsrickert) [inline packages/sit-onyx/playwright/snapshots/components/OnyxIconButton/button-state--disabled-variation--secondary-focusState--active-chromium-linux.png:1]: @jannick-ux Imho the contrast for the disabled icon button is very low...

File: packages/sit-onyx/src/components/OnyxIconButton/OnyxIconButton.vue (hunk 1 above)
Comment (JoCa96) [reply packages/sit-onyx/src/components/OnyxIconButton/OnyxIconButton.vue]: done

File: packages/sit-onyx/src/components/OnyxIconButton/OnyxIconButton.vue [hunk 3]
```diff
@@ -0,0 +1,113 @@
...
+
+const props = withDefaults(defineProps<OnyxIconButtonProps>(), {
+  disabled: false,
+  type: "button",
+  variation: "primary",
+  size: "24px",
+});
+
+defineSlots<{
+  /** Slot for an custom icon. Property `icon` will have no effect if custom content is passed. */
```
Comment (larsrickert) [inline packages/sit-onyx/src/components/OnyxIconButton/OnyxIconButton.vue:14]:
```suggestion
  /** Slot for an custom icon. Property `icon` must be omitted. */
```

The v-if checks for the existence of the icon prop, not the slot

**Review by larsrickert — CHANGES_REQUESTED** [review]

File: packages/sit-onyx/src/components/OnyxIconButton/OnyxIconButton.vue (hunk 3 above)
Comment (JoCa96) [reply packages/sit-onyx/src/components/OnyxIconButton/OnyxIconButton.vue:14]: true

Comment (JoCa96) [reply packages/sit-onyx/src/components/OnyxIconButton/OnyxIconButton.vue]: updated

**Review by larsrickert — APPROVED** [review]

File: packages/sit-onyx/playwright/snapshots/components/OnyxIconButton/button-state--disabled-variation--secondary-focusState--active-chromium-linux.png
Comment (JoCa96) [reply packages/sit-onyx/playwright/snapshots/components/OnyxIconButton/button-state--disabled-variation--secondary-focusState--active-chromium-linux.png:1]: Created a follow-up ticket: [#577](https://github.com/SchwarzIT/onyx/issues/577)

**Pull request merged**